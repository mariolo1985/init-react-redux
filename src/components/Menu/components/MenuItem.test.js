import React from 'react';
import { render, shallow } from 'enzyme';
import { MenuItem } from '..';

describe('Footer component', () => {
    let menuItemRenderCopy;
    let menuItemShallowCopy;
    beforeEach(() => {
        const item = {
            name: 'Services 1',
            tagline: 'Here is what we do',
            url: '/services1.html'
        };

        menuItemRenderCopy = render(
            <MenuItem item={item} />
        );

        menuItemShallowCopy = shallow(
            <MenuItem item={item} />
        );
    });

    describe('Initial Render', () => {
        it('Renders correctly', () => {
            expect(menuItemRenderCopy.find(MenuItem)).toMatchSnapshot();
        });
    });

    describe('Shallow Render', () => {
        it('Renders correctly', () => {
            expect(menuItemShallowCopy.find(MenuItem)).toMatchSnapshot();
        });
    });
});
