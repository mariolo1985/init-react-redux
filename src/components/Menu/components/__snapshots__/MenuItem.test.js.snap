// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Initial Render Renders correctly 1`] = `
initialize {
  "_root": initialize {
    "0": Object {
      "attribs": Object {
        "class": "menu-item",
      },
      "children": Array [
        Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Here is what we do",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "menu-item-link",
              "href": "/services1.html",
            },
            "children": Array [
              Object {
                "data": "Services 1",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
            },
          },
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        Object {
          "attribs": Object {
            "class": "menu-item-link",
            "href": "/services1.html",
          },
          "children": Array [
            Object {
              "data": "Services 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Here is what we do",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
          },
        },
      ],
      "name": "li",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "normalizeWhitespace": false,
      "withDomLvl1": true,
      "xml": false,
    },
  },
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
  "prevObject": initialize {
    "0": Object {
      "attribs": Object {
        "class": "menu-item",
      },
      "children": Array [
        Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Here is what we do",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "menu-item-link",
              "href": "/services1.html",
            },
            "children": Array [
              Object {
                "data": "Services 1",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
            },
          },
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        Object {
          "attribs": Object {
            "class": "menu-item-link",
            "href": "/services1.html",
          },
          "children": Array [
            Object {
              "data": "Services 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Here is what we do",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
          },
        },
      ],
      "name": "li",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "normalizeWhitespace": false,
      "withDomLvl1": true,
      "xml": false,
    },
  },
}
`;

exports[`Footer component Shallow Render Renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <MenuItem
      item={
        Object {
          "name": "Services 1",
          "tagline": "Here is what we do",
          "url": "/services1.html",
        }
      }
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Here is what we do
          </p>,
          <a
            className="menu-item-link"
            href="/services1.html"
          >
            Services 1
          </a>,
        ],
        "className": "menu-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Here is what we do",
          },
          "ref": null,
          "rendered": "Here is what we do",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Services 1",
            "className": "menu-item-link",
            "href": "/services1.html",
          },
          "ref": null,
          "rendered": "Services 1",
          "type": "a",
        },
      ],
      "type": "li",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Here is what we do
            </p>,
            <a
              className="menu-item-link"
              href="/services1.html"
            >
              Services 1
            </a>,
          ],
          "className": "menu-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Here is what we do",
            },
            "ref": null,
            "rendered": "Here is what we do",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Services 1",
              "className": "menu-item-link",
              "href": "/services1.html",
            },
            "ref": null,
            "rendered": "Services 1",
            "type": "a",
          },
        ],
        "type": "li",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Here is what we do
          </p>,
          <a
            className="menu-item-link"
            href="/services1.html"
          >
            Services 1
          </a>,
        ],
        "className": "menu-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Here is what we do",
          },
          "ref": null,
          "rendered": "Here is what we do",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Services 1",
            "className": "menu-item-link",
            "href": "/services1.html",
          },
          "ref": null,
          "rendered": "Services 1",
          "type": "a",
        },
      ],
      "type": "li",
    },
  ],
}
`;
